typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -90,20 +93,136 @@ export class EvmCoreModule extends HyperlaneModule<
     const transactions: AnnotatedEV5Transaction[] = [];
 
     transactions.push(
-      ...this.createMailboxOwnershipTransferTx(actualConfig, expectedConfig),
+      ...(await this.createDefaultIsmUpdateTxs(actualConfig, expectedConfig)),
+      ...this.createMailboxOwnerUpdateTxs(actualConfig, expectedConfig),
     );
 
     return transactions;
   }
 
+  /**
+   * Create a transaction to update an existing ISM config, or deploy a new ISM and return a tx to setDefaultIsm
+   *
+   * @param actualConfig - The on-chain router configuration, including the ISM configuration, and address.
+   * @param expectedConfig - The expected token router configuration, including the ISM configuration.
+   * @returns Transaction that need to be executed to update the ISM configuration.
+   */
+  async createDefaultIsmUpdateTxs(
+    actualConfig: CoreConfig,
+    expectedConfig: CoreConfig,
+  ): Promise<AnnotatedEV5Transaction[]> {
+    const updateTransactions: AnnotatedEV5Transaction[] = [];
+
+    const actualDefaultIsmConfig = actualConfig.defaultIsm as DerivedIsmConfig;
+
+    // Try to update (may also deploy) Ism with the expected config
+    const { deployedIsm, ismUpdateTxs } = await this.deployOrUpdateIsm(
+      actualDefaultIsmConfig,
+      expectedConfig.defaultIsm,
+    );
+
+    if (ismUpdateTxs.length) {
+      updateTransactions.push(...ismUpdateTxs);
+    }
+
+    const newIsmDeployed = actualDefaultIsmConfig.address !== deployedIsm;
+    if (newIsmDeployed) {
+      const { mailbox } = this.serialize();
+      const contractToUpdate = Mailbox__factory.connect(
+        mailbox!,
+        this.multiProvider.getProvider(this.domainId),
+      );
```

Comment:
seems like a `this.mailbox` would be a useful util

Reply:
+